### ISSUE TITLE
Used Parameter reported as unused

### ISSUE THREAD

**Issue by vandmo** [awslabs/cfn-python-lint#379]
*cfn-lint version: 0.7.3*

Not sure if AWS::Serverless is supported but if it is then this template produces an incorrect warning:
```yaml
AWSTemplateFormatVersion: '2010-09-09'
Transform: 'AWS::Serverless-2016-10-31'
Parameters:
  Version:
    Type: String

Globals:
  Function:
    CodeUri:
      Bucket: "somebucket"
      Key: !Sub "lambda/code/lambda-${Version}-shaded.jar"

Resources:
  SomeLambda:
    Type: 'AWS::Serverless::Function'
    Properties:
      Handler: com.SomeLambda::handleRequest
      Runtime: java8
      MemorySize: 256
```
Output from cfn-lint:
```
W2001 Parameter Version not used.
bug.yaml:4:3
```

**Comment by kddejong** [comment]
thanks.  Looking into this one.  We do process the transform before doing any rule checking.  Since globals gets removed it looks like we lose the variable usage.  

I should be able to come up with some fix for this but thinking through the best way to do it.
